tern/analyze/default/container/image.py

```diff
@@ -73,3 +77,23 @@ def analyze(image_obj, options):
         passthrough.run_extension(image_obj, options.extend, options.redo)
     else:
         default_analyze(image_obj, options)
+
+
+def common_container_procedure(image, options, image_file=None):
+    """Setup the image object and anything else for analysis"""
+    # Add a Notice object for each layer
+    for layer in image.layers:
+        origin_str = 'Layer {}'.format(layer.layer_index)
+        layer.origins.add_notice_origin(origin_str)
+    # Set up working directories and mount points
+    rootfs.set_up()
+    # analyze the image
+    analyze(image, options)
+    if image_file == "Full_Image":
+        lock.set_imported_layers(image)
```

Comment:
I am a little conflicted here. On the one hand, having a set of common steps that can be used in the container image case and the dockerfile case is great. On the other hand, the `set_imported_layers` function is only used in the dockerfile case. So I would rather not include this line here.

Reply:
@nishakm I think the issue with removing `set_imported_layers` from here is that the call to set_imported_layers needs to happen before we call `common.save_to_cache` which is the next line in this new function.

Reply:
I understand. The reason I duplicated the `cimage.analyze` function in `container/run.py` and `dockerfile/run.py` is so contributors can understand where the execution path is. So `image_file` shouldn't even exist in `container/run.py`. Perhaps `image.py` needs to live in `default` so the functions can be used both in `container/run.py` and `dockerfile/run.py`? What do you think?

Reply:
@nishakm so, should I remove the line that you asked to remove? If I do so then how I would be able to set `lock.set_imported_layers(full_image)`? Any workaround for this?

Reply:
I think the way around this is to move `image.py` out of the `container` module, but I'd like @rnjudge's input as well.